WickedEngine/wiECS.h

```diff
@@ -346,6 +406,103 @@ namespace wi::ecs
 		// Disallow this to be copied by mistake
 		ComponentManager(const ComponentManager&) = delete;
 	};
+
+	// This is the class to store all ComponentManagers,
+	// this is useful for bulk operation of all attached components within an entity
+	class ComponentLibrary
+	{
+	public:
+		wi::unordered_map<uint32_t, std::unique_ptr<ComponentManager_Interface>> componentManagers;
+
+		// Create an instance of ComponentManager of a certain data type
+		// Once added, cannot be removed!
+		template<typename T> inline ComponentManager<T>& Register(std::string name){
+			uint64_t name_hashed = wi::helper::string_hash(name.c_str());
+			componentManagers[name_hashed] = std::make_unique<ComponentManager<T>>();
+			return static_cast<ComponentManager<T>&>(*componentManagers[name_hashed]);
+		}
+
+
+		// Bulk operations
+		// Perform deep copy of all the contents of "other" into this
+		inline void Copy(const ComponentLibrary& other){
```

Comment:
Can you make `{` start on new line because that's what most of the other engine code follows?